pkg/schema/types.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2019. PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package schema contains the type to generate the openapiv3 schema for TiDB cluster config
+// +groupName=schema.pingcap.com
+// +versionName=v1alpha1
+package schema
+
+import (
+	"github.com/pingcap/tidb-operator/pkg/apis/pingcap/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +kubebuilder:resource
```

Comment:
`controller-gen` do not support filtering resources in the same package, create a dummy type to generate the configuration schema.